controllers/nodehealthcheck_controller.go

```diff
@@ -95,7 +95,16 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return result, err
 	}
 
-	// check if we need to disable NHC because of existimg MHCs
+	// check if we need to patch status before we exit Reconcile
+	nhcOrig := nhc.DeepCopy()
+	defer func() {
```

Comment:
Why have we moved patching of the status to the beginning of the reconcile rather than the end (as it was before)?

Reply:
It's still executed at the end, see https://go.dev/tour/flowcontrol/12
Reason for this change is: I want to avoid having multiple status patches during reconcile. Now I only modify status without running patch, and `defer()` ensures that patch is called when reconcile is done.

Reply:
Oho cool! _defer_  seems like a very elegant way to execute something at the end.